packages/edit-site/src/components/dataviews/sidebar-content.js

```diff
@@ -1,4 +1,205 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalItemGroup as ItemGroup,
+	DropdownMenu,
+	MenuGroup,
+	MenuItem,
+	Modal,
+	Button,
+	TextControl,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { moreVertical, plus, page, columns } from '@wordpress/icons';
+import { __ } from '@wordpress/i18n';
+import { useContext, useState, useEffect, useMemo } from '@wordpress/element';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+
+/**
+ * Internal dependencies
+ */
+import SidebarNavigationItem from '../sidebar-navigation-item';
+import DataviewsContext from '../dataviews/context';
+import { DEFAULT_VIEWS } from './provider';
+
+function DataViewItem( { dataviewId, setCurrentViewId, isActive } ) {
+	const { dataview } = useSelect(
+		( select ) => {
+			const { getEditedEntityRecord } = select( coreStore );
+			return {
+				dataview: getEditedEntityRecord(
+					'postType',
+					'wp_dataviews',
+					dataviewId
+				),
+			};
+		},
+		[ dataviewId ]
+	);
+	const icon = useMemo( () => {
+		const viewContent = JSON.parse( dataview.content );
+		switch ( viewContent.type ) {
+			case 'list':
+				return page;
+			case 'grid':
+				return columns;
+		}
+	}, [ dataview.content ] );
+	return (
+		<SidebarNavigationItem
+			icon={ icon }
+			onClick={ () => {
+				setCurrentViewId( dataviewId );
+			} }
+			aria-current={ isActive ? 'true' : undefined }
+			suffix={
+				<DropdownMenu
+					icon={ moreVertical }
+					label={ __( 'Actions' ) }
+					toggleProps={ {
+						style: {
+							color: 'inherit',
+						},
+					} }
+				>
+					{ ( { onClose } ) => (
+						<MenuGroup>
+							<MenuItem onClick={ () => onClose() }>
+								{ __( 'Rename' ) }
+							</MenuItem>
+							<MenuItem onClick={ () => onClose() }>
+								{ __( 'Delete' ) }
+							</MenuItem>
+						</MenuGroup>
+					) }
+				</DropdownMenu>
+			}
+		>
+			{ dataview.title }
+		</SidebarNavigationItem>
+	);
+}
+
+function AddNewItem( { type, taxonomyId } ) {
+	const { saveEntityRecord } = useDispatch( coreStore );
+
+	const [ isAdding, setIsAdding ] = useState( false );
+	const [ title, setTitle ] = useState( '' );
+	const { dataViewIsSaving } = useSelect( ( select ) => {
+		const { isSavingEntityRecord } = select( coreStore );
+		return {
+			dataViewIsSaving: isSavingEntityRecord(
+				'postType',
+				'wp_dataviews'
+			),
+		};
+	}, [] );
+	useEffect( () => {
```

Comment:
Can we avoid this effect by using "await" in the event handler instead?
Also, why are we "saving" on rename automatically, can't we just "edit" the entity and let the user save later. I guess this is more of a design question (and should probably match what we do in all screens). cc @jameskoster @SaxonF

Reply:
This is not for renaming, this is to create a new entity, the user provides the name of the entity, and we need to create it so then the user can change that entity that did not exist before.

Reply:
Saxon has thought more about this so I'll defer to him, but my initial reaction is to wonder why we need the user to actively save views. Could they be auto-saved instead? Requiring a save makes the UX a bit long-winded, and the mixture of site + admin details in the save modal feels a little off to me.

Reply:
Agree with Riad here. It will be a bit cleaner with await, no? Something similar to `AddCustomGenericTemplateModalContent`.